src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller.cs

```diff
@@ -108,13 +110,17 @@ private IList<ISignatureHelpProvider> GetProviders()
         {
             this.AssertIsForeground();
 
-            if (_providers == null)
+            var snapshot = this.SubjectBuffer.CurrentSnapshot;
```

Comment:
Why grab the snapshot? You can just do this.SubjectBuffer.ContentType directly and then skip the locals.